dace/codegen/targets/fpga.py

```diff
@@ -1848,7 +1848,7 @@ def get_next_scope_entries(self, sdfg, dfg, scope_entry):
         parent_scope = dfg.scope_subgraph(parent_scope_entry)
 
         # Get all scopes from the same level
-        all_scopes = [node for node in parent_scope.topological_sort() if isinstance(node, dace.sdfg.nodes.EntryNode)]
+        all_scopes = [node for node in parent_scope.bfs_nodes() if isinstance(node, dace.sdfg.nodes.EntryNode)]
```

Comment:
why is this not `.nodes()`?

Reply:
hmm, a bit of a guess, but maybe because of the next line:

```Python
return all_scopes[all_scopes.index(scope_entry) + 1:]
```

```.nodes()``` might return a different order.